src/projectM-sdl/pmSDL.cpp

```diff
@@ -281,6 +269,13 @@ void projectMSDL::addFakePCM() {
 void projectMSDL::resize(unsigned int width_, unsigned int height_) {
     width = width_;
     height = height_;
+
+		// Hide cursor if window size equals desktop size
+	  SDL_DisplayMode dm;
+	  if (SDL_GetDesktopDisplayMode(0, &dm) == 0) {
```

Comment:
I think this just checks if the call didn't fail

Reply:
That's right. But only if the call is successful we can use the `dm` structure's members, which means if the call fails, we simply don't compare the size and carry on as before, only resizing the window.

Reply:
I don't understand why it would be expected to fail and the docs are not enlightening on this matter https://wiki.libsdl.org/SDL_GetDesktopDisplayMode
And we aren't using the `dm` structure's members.
So I still don't understand the logic here

Reply:
Only way it can fail is if SDL_video is uninitialized or the current display index is invalid:
https://github.com/libsdl-org/SDL/blob/402b86f2a88b5ee3e112dbde2ec232ee5f36572f/src/video/SDL_video.c#L141

Inj this case, `dm` will be left uninitialized and accessing the members would return random values, causing the mouse pointer to stay on screen. Doesn't hurt to check the return value as suggested by the SDL docs though.